src/main/java/hudson/scm/PerJobCredentialStore.java

```diff
@@ -105,8 +112,24 @@ private Object writeReplace() {
     private static final ThreadLocal<Boolean> IS_SAVING = new ThreadLocal<Boolean>();
 
     /*package*/ void migrateCredentials(SubversionSCM.DescriptorImpl descriptor) throws IOException {
+        CredentialsStore store = CredentialsProvider.lookupStores(project).iterator().next();
         for (Map.Entry<String, Credential> e : credentials.entrySet()) {
-            descriptor.migrateCredentials(project, e.getKey(), e.getValue());
+            StandardCredentials credential =  descriptor.migrateCredentials(store, e.getKey(), e.getValue());
+            int i = 0;
+            boolean found = false;
+            ModuleLocation[] locations = ((SubversionSCM) project.getScm()).getLocations();
```

Comment:
That's whyt it's NIT :)
A good practice would be to check the class and to throw IllegalState exception. Otherwise you'll get an issue from FindBugs